### DISCUSSION TITLE
Bad Practices with NextJS Cache

### DISCUSSION THREAD

**Discussion by HarleySalas** [medusajs/nextjs-starter-medusa#351] [General]
Are you guys aware that next tags are for server-side data cache only?

When looking at the starter code, it seems to me, that there is an assumption that cache tags are only for the browser, or scoped to the user via headers.

Take the following example:
```
export const getCustomer = cache(async function () {
  return await sdk.store.customer
    .retrieve({}, { next: { tags: ["customer"] }, ...getAuthHeaders() })
    .then(({ customer }) => customer)
    .catch(() => null)
})
```

Are you aware that you are tagging **all customers** with the same tag?
When you `revalidateTag` with the argument of `customer`, you are invalidating the data **for all users**.

**Comment by VariableVic** [comment]
Hey, thanks for the post! Yes, we're aware - this pattern was wrongfully copied from the Vercel commerce Starter. I've implemented dynamic cache tags in our recent B2B Starter and will introduce the same logic here soon.
